google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -3364,4 +3418,25 @@ public void testUploadWithEncryption() throws Exception {
     byte[] readBytes = blob.getContent(Blob.BlobSourceOption.decryptionKey(KEY));
     assertArrayEquals(BLOB_BYTE_CONTENT, readBytes);
   }
+
+  @Test
+  public void testGetNotification() {
+    Notification actualNotification =
+        storage.getNotification(BUCKET, notification.getGeneratedId());
+    assertEquals(CUSTOM_ATTRIBUTES, actualNotification.getCustomAttributes());
+    assertEquals(PAYLOAD_FORMAT.name(), actualNotification.getPayloadFormat().name());
+    assertTrue(actualNotification.getTopic().contains(TOPIC));
+  }
+
+  @Test
+  public void testListNotification() {
+    List<Notification> notifications = storage.listNotifications(BUCKET);
+    for (Notification actualNotification : notifications) {
+      if (actualNotification.getGeneratedId().equals(notification.getGeneratedId())) {
+        assertEquals(CUSTOM_ATTRIBUTES, actualNotification.getCustomAttributes());
+        assertEquals(PAYLOAD_FORMAT.name(), actualNotification.getPayloadFormat().name());
+        assertTrue(actualNotification.getTopic().contains(TOPIC));
+      }
+    }
+  }
```

Comment:
Perhaps the following test scenario will cover the notification behavior better:

- check that listNotification() returns empty list
- add new notification and check:
   * id is not null
   * other fields are expected
   * listNotification includes just created one
   * getNotification(id) returns it
- deleteNotification() and then check:
   * listNotification() returns empty list
   * getNotification(id) returns null

Reply:
done